test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/Common/Server/EndToEnd/CommonEndToEndDataSource.cs

```diff
@@ -6,19 +6,48 @@
 //------------------------------------------------------------------------------
 
 using Microsoft.Spatial;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 namespace Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd
 {
     public class CommonEndToEndDataSource
     {
-        static CommonEndToEndDataSource()
+        private static CommonEndToEndDataSource? instance;
+
+        public static CommonEndToEndDataSource Instance
+        {
+            get
+            {
+                instance ??= new CommonEndToEndDataSource();
+
+                return instance;
+            }
+        }
+
+        public CommonEndToEndDataSource()
         {
+            ResetData();
             InitializeData();
         }
 
-        public static void InitializeData()
+        public CommonEndToEndDataSource CreateCopy()

```

Comment:
Hopefully all types are serializable... Is it difficult to write a test to verify that the object returned by `CreateCopy` contains the expected data?

Reply:
Would it make better sense to remove the static `Instance` property and instead rename this method `CreateInstance` such that it's what is called when someone want to create an instance of the `CommonEndToEndDataSource`?

Reply:
The spatial types might not be serializable but in those cases we can create a different data source for them.

Reply:
I think the spatial types might not be serializable but in those cases then we can have a different data source for them i.e if there are any tests on these types.

Reply:
` ``c#  public static DefaultDataSource CreateInstance()
        {
            return new DefaultDataSource();
        }`

Reply:
https://github.com/OData/odata.net/pull/3042#discussion_r1726596861

That is what I had in mind. Otherwise, how will people know when it's okay to use the `Instance` property and when it's necessary to call `CreateCopy`?

Curious, does the class expose any other methods publicly?

Reply:
No. I know some of the methods are public but they can be private.

Reply:
updated this.